runtime/list.c

```diff
@@ -21,13 +21,20 @@ static inline bool listDescAllowsTaggedAt(ListDescPtr ldp, int64_t index, Tagged
     return memberTypeContainsTagged(mt, val);
 }
 
-ListPtr _bal_list_construct(ListDescPtr desc, int64_t capacity) {
+ListPtr _bal_list_generic_construct(ListDescPtr desc, int64_t capacity) {
```

Comment:
`_generic_` isn't really the right word in this. You use it for int arrays also, right? It's for anything that has members of size 8.

There's a bit of a mismatch between `generic` as used in the ListDesc (which means any list) and `generic` in these `list_generic` functions, where it means tagged ptr.

Reply:
Ya, I was considering `_bal_list_construct_64` and `_bal_list_construct_8`.
`generic` in `GenericArray` means "not specific", and `generic` in `list_generic` means tagged.

Reply:
I think on the C side, it will make more sense to do this in terms of bytes (i.e. sizeof) rather than bits, so _8 and _1.

I think we should probably change list_generic to list_tagged.  But let's do that in a followup PR.